finality-aleph/src/base_protocol/config.rs

```diff
@@ -0,0 +1,54 @@
+use std::iter;
+
+use array_bytes::bytes2hex;
+use sc_network::{
+    config::{NonDefaultSetConfig, NonReservedPeerMode, NotificationHandshake, Role, SetConfig},
+    NotificationService,
+};
+use sc_network_common::sync::message::BlockAnnouncesHandshake;
+use sp_core::H256;
+use sp_runtime::traits::{Block, Header};
+
+use crate::{BlockHash, BlockNumber};
+
+/// Generate a config for the base protocol and the notification service that should be passed to its service.
+pub fn generate_config<B>(
+    genesis_hash: B::Hash,
+) -> (NonDefaultSetConfig, Box<dyn NotificationService>)
```

Comment:
Consider using struct here instead of tuple + add some alias for `Box<....>` pls

Reply:
When removing gossip I've solved this by returning our type that consumes the `Box<dyn ...>` - don't you want to do something similar here?

Reply:
Our `base_protocol`'s `Service` takes a raw `Box` of this form, and a lot of other arguments that it would be inconvenient and confusing to pass in here.

Reply:
Feel free to resolve but without different function name I would get confused what are those things returned from here.